bin/improver-regrid

```diff
@@ -95,15 +96,8 @@ def main():
     source_data = load_cube(args.source_data_filepath)
     target_grid = load_cube(args.target_grid_filepath)
 
-    if source_data.dtype == np.float64:
-        source_data.data = source_data.data.astype(np.float32)
-    if target_grid.dtype == np.float64:
-        target_grid.data = target_grid.data.astype(np.float32)
-
-    print("source_data0 = ", source_data)
-    print("source_data0.dtype = ", source_data.dtype)
-    print("target_grid0 = ", target_grid)
-    print("target_grid0.dtype = ", target_grid.dtype)
+    check_cube_not_float64(source_data, fix=True)
```

Comment:
Note, we will have a follow on ticket to consider if there is a better way of doing this, but this enables us to test the suite changes for now.